server/src/com/thoughtworks/go/config/GoConfigDao.java

```diff
@@ -624,7 +624,42 @@ public CruiseConfig update(CruiseConfig cruiseConfig) throws Exception {
         }
     }
 
+    public static class UpdateEnvironmentCommand implements UpdateConfigCommand {
+
+        private final String environmentToUpdate;
+        private final EnvironmentConfig newEnvironmentConfig;
+
+        public UpdateEnvironmentCommand(String environmentToUpdate, EnvironmentConfig newEnvironmentConfig) {
```

Comment:
Do you have a test for this behaviour, so that the next person who decides to "refactor" this code in the future knows not to?